spec/graphql/queries/requests/encounters_index_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+module Queries
+  module Encounters
+    RSpec.describe Encounters, type: :request do
+      describe 'encounters' do
+        before(:each) do
+          @list = create_list(:encounter_monster, 6)
+        end
+        it 'returns all encounters associated with a given User ID' do
+          post '/graphql', params: { query: query, variables: { 'userName': "Drizzt" }}
+          json = JSON.parse(response.body)
+          data = json['data']['encounters']
+          expect(data.count).to eq(6)
+          encounter = data.first
+          
+          expect(encounter['userName']).to eq("Drizzt")
+          expect(encounter['encounterName']).to be_a(String)
+          expect(encounter['partySize']).to be_a(Integer)
+          expect(encounter['partyLevel']).to be_a(Integer)
+          expect(encounter['summary']).to be_a(String)
+          expect(encounter['description']).to be_a(String)
+          expect(encounter['treasure']).to be_a(String)
+          expect(encounter['encounterMonsters']).to be_a(Array)
+        end
+      end
+      def query
+        <<~GQL
+          query getEncounters($userName: String!) {
+            encounters(userName: $userName) {
+              userName
+              encounterName
+              partySize
+              partyLevel
+              summary
+              description
+              treasure
+              encounterMonsters {
+                monsterName
+              }
```

Comment:
This is good to test all we provide, unless we want to pair down the data to be less if need be, but just a note based on what frontend is using from this particular endpoint.

They really are only displaying `encounterName`, `partySize`, `partyLevel`, `summary`, and list of `encounterMonsters`

Reply:
I think for testing we should be covering the full amount of data, since customization really relies on how FE wants to write there queries, as opposed to what we make available to query. Even if we cut down on what we query in the test it wouldn't show any functional change in the code, would it?

Reply:
This kinda goes along with my other comment, just marking it as a clean up item for later and a bit of a semantic item. If it is provided yes we should test. But LATER if we remove from the `queryType` we can remove the tests.

You are right and It does not functionally change too much, as the frontend consumption has control over what is queried because we are using GraphQL. But I generally think in terms of, we only provide what is neccesary, and frontend is not going to be using some of this data.

Reply:
You are correct that the only things we are displaying on the preview cards are `encounterName`, `partySize`, `partyLevel`, `summary`, and list of `encounterMonsters`. It's not really a problem to include other things though, because we could always decide to render more items later. Also, I know we need a LOT more data for the Builder and Details, but I think that's a different endpoint, right?

Reply:
I think we might actually prefer to get the larger information from this endpoint, but @tenthwalker can correct me if I'm wrong since they've been working on the detail page.

Reply:
Clarified in Slack